boards: adafruit: add initial support for esp32 feather v2

lvoytek opened the pull request:
The Adafruit ESP32 Feather V2 is a board that uses the Feather standard layout. It uses an ESP32-PICO v3 02 module, and includes a USB-C connector, LiPo battery charger, NeoPixel RGB LED, and STEMMA QT connector.

Not to be confused with the adafruit esp32s3 feather, which I noticed was [just added](https://github.com/zephyrproject-rtos/zephyr/pull/75844) a few days ago :)

kartben commented on boards/adafruit/feather_esp32_v2/doc/img/adafruit_feather_esp32_v2.jpg:
feel free to directly use the attached -- it has transparent background, too :)
[adafruit_feather_esp32.webp.zip](https://github.com/user-attachments/files/19282540/adafruit_feather_esp32.webp.zip)

lvoytek replied:
Thanks for creating that! Added

lvoytek committed:
boards: adafruit: add initial support for esp32 feather v2

The Adafruit ESP32 Feather V2 is a board that uses the Feather standard
layout. It uses an ESP32-PICO v3 02 module, and includes a USB-C
connector, LiPo battery charger, NeoPixel RGB LED, and STEMMA QT
connector.

Signed-off-by: Lena Voytek <lena@voytek.dev>

lvoytek commented on boards/adafruit/feather_esp32_v2/board.yml:
```diff
@@ -0,0 +1,6 @@
+board:
+  name: adafruit_feather_esp32_v2
+  full_name: Adafruit ESP32 Feather V2
```
Updated naming and added a single default "2" revision. I know it's deprecated but if there's interest I could also port V1 since I still own several of them :)

lvoytek commented on boards/adafruit/feather_esp32_v2/adafruit_feather_esp32_v2-pinctrl.dtsi:
```diff
@@ -0,0 +1,61 @@
+			pinmux = <SPIM2_MISO_GPIO21>,
+				 	 <SPIM2_SCLK_GPIO5>;
+		};
+		group2 {
```
done

sylvioalves commented on boards/adafruit/feather_esp32/adafruit_feather_esp32_appcpu.yaml:
```diff
@@ -0,0 +1,27 @@
+identifier: adafruit_feather_esp32/esp32/appcpu
+name: Adafruit ESP32 Feather
```
Would you mind adding `APPCPU` and `PROCPU` accordingly in both yaml files?
`name: Adafruit ESP32 Feather APPCPU`

lvoytek replied:
Done, added to the dts files too

sylvioalves commented on boards/adafruit/feather_esp32/adafruit_feather_esp32_procpu.dts:
```diff
@@ -0,0 +1,150 @@
+
+&esp32_bt_hci {
+	status = "okay";
+};
```
I suggest adding thw wifi node as well:

```c
&wifi {
	status = "okay";
};
```

lvoytek replied:
That's a great point, added :)

sylvioalves commented on boards/adafruit/feather_esp32/adafruit_feather_esp32_procpu.yaml:
```diff
@@ -0,0 +1,15 @@
+  - watchdog
+  - uart
+  - pinmux
+  - nvs
```
Should it support those feather entries? (and peripherals..)
```c
  - can
  - counter
  - entropy
  - pwm
  - input
  - feather_serial
  - feather_i2c
  - feather_spi
  ```

sylvioalves commented on boards/adafruit/feather_esp32/doc/index.rst:
```diff
@@ -0,0 +1,108 @@
+- NeoPixel RGB LED
+- Reset and user buttons
+- STEMMA QT I2C connector
+
```
Would be good do add the supported peripherals in the documentation as well:

```
Supported Features
==================

.. zephyr:board-supported-hw::
```

lvoytek commented:
Thanks for the review @sylvioalves! Updated based on your comments

kartben commented on boards/adafruit/feather_esp32/board.yml:
```diff
@@ -0,0 +1,11 @@
+board:
+  name: adafruit_feather_esp32
+  full_name: Adafruit ESP32 Feather
```
Nit: all the other supported Feather boards are named "Adafruit Feather XXX" 

```suggestion
  full_name: Adafruit Feather ESP32
```

lvoytek replied:
That's fair. I originally named it based on the adafruit.com listing but now I see all ESP32 feathers there now use the "Adafruit ESP32... Feather" convention. It's definitely better to have naming consistency here, so I've updated the name in the yml and dts files :)
